src/expr/tests/test_runner.rs

```diff
@@ -178,6 +179,49 @@ mod test {
         Ok(equivalences)
     }
 
+    fn test_interpret(s: &str) -> Result<Vec<String>, String> {
+        let mut input_stream = tokenize(s)?.into_iter();
```

Comment:
I don't know this part of the code... am I correct in reading this function as getting random values and checking that the abstract interpretation gives correct approximations?

Reply:
Or rather, not random values, but the tests in the `src/tests/testdata/interpret`.

Reply:
Yes. More specifically, the 

```rust
"interpret" => match test_interpret(&s.input) { ... }
```
call below ensures that the tests annotated with `directive = interpret` are interpreted by `test_interpret`.